applications/http/unikernel.ml

```diff
@@ -159,7 +145,7 @@ struct
                 m
         in
         let alpn_protocols =
```

Comment:
I can't make a full suggestion thanks to GitHub limitations, but
```suggestion
        let alpn_protocols = alpn () in
```
And then delete the match

Reply:
I don't quite understand your suggestion... the value `alpn ()` will be a `string option` -- but the match only allows certain strings, fails on others, and has a default...

Reply:
Something like this is what I meant. I wrote another comment that seems to have been lost
```diff
diff --git a/applications/http/unikernel.ml b/applications/http/unikernel.ml
index 6c0b9e6..b69c6d8 100644
--- a/applications/http/unikernel.ml
+++ b/applications/http/unikernel.ml
@@ -17,8 +17,10 @@ let tls_port =
   Mirage_runtime.register_arg Arg.(value & opt int 4343 doc)
 
 let alpn =
-  let doc = Arg.info ~doc:"Protocols handled by the HTTP server." [ "alpn" ] in
-  Mirage_runtime.register_arg Arg.(value & opt (some string) None doc)
+  let alpns = [ "h2"; "http/1.1" ] in
+  let doc = Printf.sprintf "Protocols handled by the HTTP server. Must be %s." (Arg.doc_alts alpns) in
+  let doc = Arg.info ~doc [ "alpn" ] in
+  Mirage_runtime.register_arg Arg.(value & opt_all (enum (List.map (fun v -> (v, v)) alpns)) alpns doc)
 
 let ( <.> ) f g x = f (g x)
 let always x _ = x
@@ -148,12 +150,7 @@ struct
                  private key. Received error %s."
                 m
         in
-        let alpn_protocols =
-          match alpn () with
-          | None -> [ "h2"; "http/1.1" ]
-          | Some (("http/1.1" | "h2") as proto) -> [ proto ]
-          | Some proto -> Fmt.failwith "Invalid ALPN protocol %S" proto
-        in
+        let alpn_protocols = alpn () in
         match Tls.Config.server ~certificates ~alpn_protocols () with
         | Error (`Msg m) -> Fmt.failwith "TLS configuration error: %s." m
         | Ok tls -> tls
```

Reply:
ah, that makes sense. feel free to push such a commit to this PR